src/core/diags/factory_diags.cpp

```diff
@@ -74,8 +74,7 @@ void Diags::ProcessChannel(int aArgCount, char *aArgVector[], char *aOutput, siz
     SuccessOrExit(error = ParseLong(aArgVector[0], value));
     VerifyOrExit(value >= Radio::kChannelMin && value <= Radio::kChannelMax, error = OT_ERROR_INVALID_ARGS);
 
-    mChannel = static_cast<uint8_t>(value);
-    otPlatDiagChannelSet(mChannel);
+    otPlatDiagChannelSet(static_cast<uint8_t>(value));
```

Comment:
We don't need to save it in `mChannel`? How is `mChannel` in `Diags` is used? Do we need this variable?